### PR TITLE
support send response data to dingtalk, feishu, business wechat webho…

### PR DESCRIPTION
**Pull request by linversion**
…ok and also support send custom text

Describe what problem this pull request solves and how to solve it. If there is an issue associated with it, please attach a link.

添加了发送request body、request header、response到飞书、钉钉、企业微信机器人的功能，也可以发送自己输入的信息，提高前后端联调效率。

## Pull Request Checklist

- [ ] I have read the [UME contribution document](../CONTRIBUTING.md) and understand how to contribute, commit the code according to the rules. 我已阅读过 UME 贡献文档，并了解如何进行贡献，按照规则提交了代码
- [ ] I have added the necessary comments in code to ensure that other contributors can understand the reason for the change. 我在修改中已经添加了必要的注释，以确保便于其他贡献者理解修改原因
- [ ] The code has been formatted by dartfmt before push. 代码在提交前已经经过 dartfmt 进行了格式化
- [ ] Change does not involve the adjustment of test cases. Or all existing and new tests are passing. 改动不涉及测试用例调整，或 example 工程与单元测试已经完全跑通

If you need help, consider [Join ing the ByteDance Flutter Exchange Group](https://applink.feishu.cn/client/chat/chatter/add_by_link?link_token=67au2f75-3783-41b0-8868-0fc0178f1fd8).

如有任何问题，可以[加入字节跳动 Flutter 交流群](https://applink.feishu.cn/client/chat/chatter/add_by_link?link_token=67au2f75-3783-41b0-8868-0fc0178f1fd8)。

Or contact [author](mailto:sunkai.dev@bytedance.com).

或随时[联系开发者](mailto:sunkai.dev@bytedance.com)。

### CODE REVIEW

File: kits/flutter_ume_kit_dio/lib/src/net/dio_manager.dart [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+typedef RequestBodyBuilder = Map<String, dynamic> Function(String content);
+const weiXin = 'https://qyapi.weixin.qq.com';
+const dingTalk = 'https://oapi.dingtalk.com';
+const feiShu = 'https://open.feishu.cn';
+
+class DioManager {
+  static DioManager? _instance;
+
+  late Dio _dio;
+  // 是否使用发送到webhook功能
```
Comment (talisk) [inline kits/flutter_ume_kit_dio/lib/src/net/dio_manager.dart]: Please write code comments in English.

File: kits/flutter_ume_kit_dio/lib/src/net/dio_manager.dart [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      connectTimeout: 15000,
+      receiveTimeout: 15000,
+      headers: {'Content-Type': Headers.jsonContentType},
+    );
+    _dio = Dio(options);
+  }
+
+  static DioManager get instance => _instance ??= DioManager._internal();
+
+  // 自定义请求体构造器
```
Comment (talisk) [inline kits/flutter_ume_kit_dio/lib/src/net/dio_manager.dart]: Please write code comments in English.

File: kits/flutter_ume_kit_dio/lib/src/net/dio_manager.dart [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    useBot = true;
+  }
+
+  void setWebhookUrl(String? url) {
+    if (url == null) return;
+    _dio.options.baseUrl = url;
+    initRequestBodyBuilder(url);
+  }
+
+  // 发送response到webhook
```
Comment (talisk) [inline kits/flutter_ume_kit_dio/lib/src/net/dio_manager.dart]: Please write code comments in English.

File: kits/flutter_ume_kit_dio/lib/src/net/dio_manager.dart [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+          "\nresponse body: \n$responseBody\n\nrequest header\n$requestHeader";
+      await _dio.post('', data: requestBodyBuilder!(content));
+      return true;
+    } catch (error) {
+      debugPrint('error on send response: $error');
+      return false;
+    }
+  }
+
+  // 发送自定义信息到webhook
```
Comment (talisk) [inline kits/flutter_ume_kit_dio/lib/src/net/dio_manager.dart]: Please write code comments in English.

File: kits/flutter_ume_kit_dio/lib/src/net/dio_manager.dart [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  Future<void> sendCustomText(String content) async {
+    try {
+      await _dio.post('', data: requestBodyBuilder!('not dio:\n $content'));
+    } catch (error) {
+      debugPrint('error on send response: $error');
+    }
+  }
+
+  // 构造对应webhook的请求体格式
```
Comment (talisk) [inline kits/flutter_ume_kit_dio/lib/src/net/dio_manager.dart]: Please write code comments in English.

File: kits/flutter_ume_kit_dio/lib/src/net/dio_manager.dart [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+  void initRequestBodyBuilder(String url) {
+    if (url.startsWith(dingTalk) || url.startsWith(weiXin)) {
+      //钉钉或微信机器人
```
Comment (talisk) [inline kits/flutter_ume_kit_dio/lib/src/net/dio_manager.dart]: Please write code comments in English.

File: kits/flutter_ume_kit_dio/lib/src/net/dio_manager.dart [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+      requestBodyBuilder = (content) => {
+            "msgtype": "text",
+            "text": {"content": content}
+          };
+      useBot = true;
+    } else if (url.startsWith(feiShu)) {
+      //飞书机器人
```
Comment (talisk) [inline kits/flutter_ume_kit_dio/lib/src/net/dio_manager.dart]: Please write code comments in English.

File: kits/flutter_ume_kit_dio/lib/src/widgets/custom_message_page.dart [hunk 8]
```diff
@@ -0,0 +1,51 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_ume_kit_dio/src/net/dio_manager.dart';
+
+// 自定义信息输入页面
```
Comment (talisk) [inline kits/flutter_ume_kit_dio/lib/src/widgets/custom_message_page.dart]: Please write code comments in English. And the comment before a public API should start with `///`. It indicate that it's a documentation comment.

File: kits/flutter_ume_kit_dio/lib/src/widgets/pluggable_state.dart [hunk 9]
```diff
@@ -384,3 +411,25 @@ extension _DateTimeExtension on DateTime {
...
+//             1}),
+//             data: {
+//               'total': 1,
+//               'items': [{'name': 'test', 'time': DateTime.now().millisecondsSinceEpoch}]
+//             },
+//         ),
+//       ),
+//     ),
+//   ));
+// }
```
Comment (talisk) [inline kits/flutter_ume_kit_dio/lib/src/widgets/pluggable_state.dart]: Is this code for testing? You should remove it.

Comment (talisk) [reply kits/flutter_ume_kit_dio/lib/src/widgets/pluggable_state.dart]: And please write example code in the `Example` project of `flutter_ume`

**Review by talisk — CHANGES_REQUESTED** [review]
Please write example code in Example project. And then I will ask @AlexV525 to review the code.

**Comment by AlexV525** [comment]
I'm not sure why an interceptor should send such content, but a log interceptor might do so. If you want to collect requests and responses, it might be better to expose the method to obtain, rather than nested the upload function.

FYI @talisk.

**Comment by linversion** [comment]
> Please write example code in Example project. And then I will ask @AlexV525 to review the code.

When you register the DioInspector, there are three parameter in it's constructor, only the dio is required. Not sure how to write the example code.

**Pull request closed without merging**